crypto/bio/bss_bio.c

```diff
@@ -635,6 +635,7 @@ static int bio_make_pair(BIO *bio1, BIO *bio2)
         b2->buf = OPENSSL_malloc(b2->size);
         if (b2->buf == NULL) {
             BIOerr(BIO_F_BIO_MAKE_PAIR, ERR_R_MALLOC_FAILURE);
+            OPENSSL_free(b1->buf);
```

Comment:
There does not seem to be a leak here at all and this change could result in a double free. The memory allocated to `b1->buf` is safe to stay associated with the `bio1` object. Ultimately when that gets freed the `buf` member will also get freed.